### ISSUE TITLE
Request: Add zlib-ng as a zlib replacement

### ISSUE THREAD

**Issue by woot000** [m-ab-s/media-autobuild_suite#2812]
Would be nice to replace zlib with zlib-ng for the performance uplift, since a lot of libraries in the suite use zlib. I'm submitting an issue instead of a pull request because I'm waiting for zlib-ng to be [added as an MSYS2 package](https://github.com/msys2/MINGW-packages/pull/20844) before I try adding it.

**Comment by L4cache** [comment]
> Specifically, it would be zlib-ng-compat as a drop-in replacement, correct?
> 
> For an optimized approach like mabs, it might not be necressary to wait for the MSYS2 package but to build mabs' own zlib-ng into local - WITH_RUNTIME_CPU_DETECTION could be disabled because it's for a specific system anyway. https://github.com/zlib-ng/zlib-ng/blob/develop/CMakeLists.txt

No, runtime cpu detection is as the name suggests, detect cpu (available instruction sets) at runtime, it's not for a specific system, disabling it will make it "system specific".
I'm talking about the general idea of runtime cpu detection, not sure if zlib-ng actually made it the other way around.

**Comment by woot000** [comment]
Just learned of another alternative, [zlib-rs](https://github.com/trifectatechfoundation/zlib-rs), which also has a zlib-compatible API and claims to be [faster than zlib-ng](https://trifectatech.org/blog/zlib-rs-is-faster-than-c/) on both compression and decompression. This looks like another good candidate to run as a zlib alternative, if and when this feature gets implemented
